### PR TITLE
MAINT: Add a branch step to the contribution guidelines.

### PR DESCRIPTION
**Pull request by oesteban**
This PR also fixes some typos and complex sentences.

### CODE REVIEW

**Comment by oesteban** [comment]
The addition of the branch step will be useful for @josephmje and @dPys when they sent PRs with the code they've developed.

BTW, I would be ready to review another PR with one more of your workflows, @josephmje. When all yours are merged, we can then start the intake of @dPys' code.

**Comment by josephmje** [comment]
To start, would the simplest workflow to PR be just connecting the anatomical and diffusion workflows and adding eddy?

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -155,76 +155,74 @@ Finally, ![GitHub labels](https://img.shields.io/github/labels/nipreps/dmriprep/
...
+    Make sure to [keep your fork up to date][link_updateupstreamwiki] with the upstream repository.<br />  
+    For example, to update your master branch on your local cloned repository:  
+      ```Shell
+      git fetch upstream
+      git checkout master
+      git merge upstream/master
+      ```
+
+1. **Create a [new branch][link_branches] to develop and maintain the proposed code changes
+  prior a pull-request (see below) is submitted and code changes are reviewed and accepted.**<br />
```
Comment (josephmje) [inline CONTRIBUTING.md]:
```suggestion
  prior to submitting a pull-request (see below) and code changes are reviewed and accepted.**<br />
```

Comment (josephmje) [reply CONTRIBUTING.md]: Just a suggestion but maybe get rid of "and code changes are reviewed and accepted"

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -155,76 +155,74 @@ Finally, ![GitHub labels](https://img.shields.io/github/labels/nipreps/dmriprep/
...
+      * `fix/<some-identifier>`: for bugfixes
+      * `enh/<feature-name>`: for new features
+      * `doc/<some-identifier>`: for documentation improvements.
+        You should name all your documentation branches with the prefix `doc/` or `docs/`
+        as that will preempt triggering the full battery of continuous integration tests.
+
+1. **Make the changes you've discussed, following the [dMRIPrep coding style guide](#dMRIPrep-coding-style-guide).**<br />
+  Try to keep the changes focused: it is generally easy to review changes that address one feature or bug at a time.
+  It can also be helpful to test your changes locally,
+  using an [dMRIPrep development environment][link_devel].
```
Comment (josephmje) [inline CONTRIBUTING.md]:
```suggestion
  using a [dMRIPrep development environment][link_devel].
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -155,76 +155,74 @@ Finally, ![GitHub labels](https://img.shields.io/github/labels/nipreps/dmriprep/
...
+  Once you are satisfied with your local changes, [add/commit/push them][link_add_commit_push]
+  to the branch on your forked repository.
+
+1. **Submit a [pull request][link_pullrequest].**<br />
+   A member of the development team will review your changes to confirm
+   that they can be merged into the main code base.<br />
+   Pull requests titles should begin with a descriptive prefix
```
Comment (josephmje) [inline CONTRIBUTING.md]:
```suggestion
   Pull request titles should begin with a descriptive prefix
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -3,22 +3,22 @@
 Welcome to the dMRIPrep repository!
 We're excited you're here and want to contribute.
 
-**Imposter syndrome disclaimer**: We want your help. No, really.
+**Imposter's syndrome disclaimer**: We want your help. No, really.
```
Comment (josephmje) [inline CONTRIBUTING.md]:
```suggestion
**Impostor syndrome disclaimer**: We want your help. No, really.
```

Comment (josephmje) [reply CONTRIBUTING.md]: Also, add the proper attribution for the Impostor syndrome disclaimer. 

Taken from the original dmriprep:

> 
> The imposter syndrome disclaimer was originally written by
> [Adrienne Lowe](https://github.com/adriennefriend) for a [PyCon
> talk](https://www.youtube.com/watch?v=6Uj746j9Heo), and was
> adapted based on its use in the README file for the [MetPy
> project](https://github.com/Unidata/MetPy).

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -92,7 +92,7 @@ the [Software Carpentry introduction to git][link_swc_intro].
 On GitHub, You'll use [Markdown][markdown] to chat in issues and pull requests.
 You can think of Markdown as a few little symbols around your text that will allow GitHub
 to render the text with a little bit of formatting.
-For example you could write words as bold (`**bold**`), or in italics (`*italics*`),
+For example, you could write words as bold (`**bold**`), or in italics (`*italics*`),
 or as a [link][rick_roll] (`[link](https://https://youtu.be/dQw4w9WgXcQ)`) to another webpage.
```
Comment (josephmje) [inline CONTRIBUTING.md]: This one isn't displaying properly.

Comment (josephmje) [reply CONTRIBUTING.md]: My mistake. Ignore this comment.

Comment (oesteban) [reply CONTRIBUTING.md]: Actually, there are two `https://https://youtu...` - although it does not affect the functionality of the link, it is completely wrong and defeats the purpose of showing how to write links. Fixed in the latest commit.

File: CONTRIBUTING.md (hunk 4 above)
Comment (oesteban) [reply CONTRIBUTING.md]: Added in a footnote.

File: CONTRIBUTING.md (hunk 1 above)
Comment (oesteban) [reply CONTRIBUTING.md]: Thanks, I have simplified this sentence and added one more step for the PR review.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -155,76 +155,80 @@ Finally, ![GitHub labels](https://img.shields.io/github/labels/nipreps/dmriprep/
...
+     * `TST`: new or updated tests ([example][tst_ex])
+     * `DOC`: new or updated documentation ([example][doc_ex])
+     * `STY`: style changes ([example][sty_ex])
+     * `REF`: refactoring existing code ([example][ref_ex])
+     * `CI`: updates to continous integration infrastructure ([example][ci_ex])
+     * `MAINT`: general maintenance ([example][maint_ex])
+     * For works-in-progress, add the `WIP` tag in addition to the descriptive prefix.
+       Pull-requests tagged with `WIP:` will not be merged until the tag is removed.
+
+1. **Have your PR reviewed by the developers team, and update your changes accordingly in your branch.**<br />
```
Comment (josephmje) [inline CONTRIBUTING.md]:
```suggestion
1. **Have your PR reviewed by the development team, and update your changes accordingly in your branch.**<br />
```

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -155,76 +155,80 @@ Finally, ![GitHub labels](https://img.shields.io/github/labels/nipreps/dmriprep/
...
+   The reviewers will take special care in assisting you address their comments, as well as dealing with conflicts
```
Comment (josephmje) [inline CONTRIBUTING.md]:
```suggestion
   The reviewers will take special care in assisting you to address their comments, as well as dealing with conflicts
```

**Pull request merged**